### PR TITLE
Add Support for Credentials

### PR DESCRIPTION
**Pull request by harkiratsm**
**Notes for Reviewers**

This PR fixes #
In meshery's settings UI, you can crud credentials, there were some proxy issues earlier caused by compromised credentials server code, I have also fixed that. 
The screenshot will be posted shortly.

image

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: server/models/remote_provider.go [hunk 1]
```diff
@@ -3467,7 +3467,8 @@ func (l *RemoteProvider) GetUserCredentials(req *http.Request, _ string, page, p
 
 	var cp CredentialsPage
 	if err = json.Unmarshal(bdr, &cp); err != nil {
-		return nil, err
+		fmt.Println("bawi is here", &cp)
```
Comment (Azanul) [inline server/models/remote_provider.go:3470]: .

**Review by Azanul — CHANGES_REQUESTED** [review]
Please fix lint issue too

File: server/models/remote_provider.go (hunk 1 above)
Comment (theBeginner86) [reply server/models/remote_provider.go:3470]: 👀

Comment (harkiratsm) [reply server/models/remote_provider.go:3470]: yeah, just some debug log.

**Comment by leecalcote** [comment]
@harkiratsm will you investigate the failing UI tests?

**Comment by harkiratsm** [comment]
@leecalcote Tests are failing on the latest as well, so I'll investigate the artifacts and create a separate PR.

**Review by theBeginner86 — APPROVED** [review]
LGTM 👌

Works great

**Review by theBeginner86 — COMMENTED** [review]
@harkiratsm 
2 followup questions:

1. Will you be adding similar view in Meshery Cloud like in Meshery UI? Icons for credential type are not visible in Meshery Cloud

image

2. Auto registration of Kubernetes credentials? That is, migrating current kubernetes contexts into credentials table and using Kubernetes as a connection.

**Review by theBeginner86 — COMMENTED** [review]
Would you please see why backend lint checks are failing?

**Review by leecalcote — CHANGES_REQUESTED** [review]
@harkiratsm question #2 above is a big question.

**Comment by leecalcote** [comment]
> @harkiratsm question #2 above is a big question.

@harkiratsm thoughts?

**Comment by harkiratsm** [comment]
The credentials are saved when the user uploads the Kubeconfig. I am pushing new changes.

Here is the screenshot.

image

**Comment by leecalcote** [comment]
Context = Connection

**Comment by Azanul** [comment]
@harkiratsm While resolving conflicts do take care that all integration tests should be passing and only one E2E test should be failing. //@MarioArriaga92

**Comment by leecalcote** [comment]
> Context = Connection

And by this, I mean to highlight that for each context defined within a kubeconfig, there should be one connection creation, maintaining a 1:1 relationship between Context and Connection.

**Comment by MarioArriaga92** [comment]
@harkiratsm not sure if this was mentioned during the call but there's a conflict in this file `ui/components/MesherySettings.js`:
CleanShot 2023-05-26 at 09 23 56@2x

**Comment by abdullah1308** [comment]
@MarioArriaga92 @theBeginner86 Any update on when this PR can be merged ?

**Comment by abdullah1308** [comment]
@harkiratsm can you resolve conflicts ?

**Comment by leecalcote** [comment]
Note that there’s a bit of formatting off with the close button being misplaced, it seems.

Screenshot 2023-06-24 at 4 47 00 PM



Also, note that one of the conflicts here steps backwards with a short compatibility list of only service meshes, not a more complete model list (which could be and should be programmatically retrieved from` /api/meshmodels/models`)

**Comment by leecalcote** [comment]
Syncing with `master`

**Comment by harkiratsm** [comment]
merging this one and will do the remaining in this PR https://github.com/meshery/meshery/pull/8035

**Comment by harkiratsm** [comment]
left over - 
1. The Type field should contain meshmodel models.

**Pull request merged**